### ISSUE TITLE
Build failure

### ISSUE THREAD

**Issue by jserv** [Kray-G/kilite#2]
I attempted to build this project on Ubuntu Linux 20.04 with the following steps:
```shell
$ git clone https://github.com/Kray-G/kilite --recursive
$ cd kilite/build
$ ./make.sh
```

However, I encountered the build failure.
```
Generating a header source code...
Building a Kilite binary...
../src/frontend/mkkir.c: In function ‘gen_binary_literal’:
../src/frontend/mkkir.c:477:9: error: a label can only be part of a statement and a declaration is not a statement
  477 |         kl_kir_opr rs = {0};
      |         ^~~~~~~~~~
../src/frontend/mkkir.c: In function ‘gen_array_literal’:
../src/frontend/mkkir.c:513:9: error: a label can only be part of a statement and a declaration is not a statement
  513 |         kl_kir_opr rs = {0};
      |         ^~~~~~~~~~
cp: cannot stat 'kilite': No such file or directory
Creating a basic library...
./make.sh: line 121: /tmp/kilite/bin/kilite: No such file or directory
Generating a static library file for gcc...
Building Kilite modules successfully ended.
```

Toolchain information:
```
$ gcc -v
...
gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 
```

**Comment by Kray-G** [comment]
Thanks for your reporting.
Okay I will check it next week.

**Comment by Kray-G** [comment]
I see it was okay with gcc 11.2.0 which is my environment by default on Linux, but it was not okay with gcc 9.4.0. BTW, it was also okay with MinGW gcc 8.1.0 on Windows.

Anyway I found the problem around a C label so I did a quick fix. It should be okay for all those above versions. Would you try it?

It was very helpful and many thanks!

**Comment by jserv** [comment]
> I see it was okay with gcc 11.2.0 which is my environment by default on Linux, but it was not okay with gcc 9.4.0. BTW, it was also okay with MinGW gcc 8.1.0 on Windows.

The previous problem was resolved with commit 8c807e9b7a94e33bb2ddd37e3144075291f45a48. Then, the linkage issue appeared.
```
Generating a header source code...
Building a Kilite binary...
/usr/bin/ld: /tmp/cc5pPD9I.o: in function `open_std_libs':
resolver.c:(.text+0x6d): undefined reference to `dlopen'
/usr/bin/ld: resolver.c:(.text+0x95): undefined reference to `dlopen'
...
/tmp/kilite/submodules/mir/mir-gen.c:5640: undefined reference to `pthread_create'
/usr/bin/ld: /tmp/kilite/submodules/mir/mir-gen.c:5642: undefined reference to `pthread_join'
/usr/bin/ld: ../bin/libmir_static.a(mir-gen.c.o): in function `MIR_gen_finish':
/tmp/kilite/submodules/mir/mir-gen.c:5709: undefined reference to `pthread_join'
/usr/bin/ld: ../bin/libminizip.a(mz_crypt_openssl.c.o): in function `sk_X509_num':
mz_crypt_openssl.c:(.text+0x14): undefined reference to `OPENSSL_sk_num'
...
```

**Comment by Kray-G** [comment]
Thanks for the report.

Unfortunately, I didn't see this phenomenon, maybe it depends on my environment, but I did work for the trial fix. The message would be the problem about a library link. Regarding `dlopen` and `pthread_*`, it will be a simple fix with link options.

However, I couldn't check it but minizip-ng seems not to be compiled with OpenSSL 3.0 with deprecated APIs. It looks related to [the issue #546 on minizip-ng](https://github.com/zlib-ng/minizip-ng/issues/546). So now I disabled OpenSSL for minizip-ng compilation. It will affect the functionality of AES encryption only. On Linux, AES encryption can't work but other functionalities should work as well as before.

And also, I will look for something like another workaround or solution for minizip-ng with OpenSSL 3.0, or introducing AES encryption with another way.

I'll report it if I have any update for that.

Thanks so much.
